### ISSUE TITLE
Feature: unused imports lint/fixer

### ISSUE THREAD

**Issue by tzachshabtay** [yoheimuta/protolint#125]
I am getting a warning from protoc but not from the linter (using the vscode extension).

For example:

`warning: Import google/protobuf/wrappers.proto but not used.`

**Comment by yoheimuta** [comment]
@tzachshabtay Thank you for your suggestion.
It sounds useful. I attached the label to this issue for starters.

**Comment by mayankbpatel** [comment]
Agree with @tzachshabtay.  It would be useful to identify unused `imports` or `messages`.

**Comment by HartBlanc** [comment]
I don't think it is possible for this lint rule to be reliably implemented in protolint?

Detection of unused imports requires the package name associated with the import to be known.
This requires reading the contents of the imported file to locate the package name definition (it can't be inferred from the import statement alone.)

As far as I can tell protolint only looks at a single proto file in isolation (as opposed to protoc which has the context of all of the files).

Please do correct me if I'm mistaken here!

**Comment by yoheimuta** [comment]
You’re right. Besides, this rule needs the `proto_path` to find imports.
If users set it in the config, then imports can be checked properly.

With some additional logic to process all imports, the rule could be implemented.

**Comment by HartBlanc** [comment]
Thanks for clarifying!

At least in my case this would make protolint much more difficult to integrate with as third party imports and the proto_path are derived in the build system and protolint (and all other linters) are run outside of this context.
